src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,433 @@
+            /// }
+            /// </summary>
+            public ImmutableArray<(IParameterSymbol parameterSymbol, TExpressionSyntax expression)> ParametersToGenerateFreshVariablesFor { get; }
+
+            /// <summary>
+            /// A dictionary that contains Parameter that should be directly replaced. Key is the parameter and Value is the replacement exprssion
+            /// It includes
+            /// 1. Parameter mapping to literal expression
+            /// 2. Parameter that has default value, and it has no argument. It should be replaced by the default value.
+            /// 3. Parameter mapping to identifier expression
+            /// Before:
+            /// void Caller(int i, int j, bool[] k)
+            /// {
+            ///     Callee(i, j, k);
+            /// }
+            /// void Callee(int a, int b, params bool[] c)
+            /// {
+            ///     DoSomething(a, b, c);
+            /// }
+            /// After:
+            /// void Caller(int i, int j, bool[] k)
+            /// {
+            ///     DoSomething(i, j, k);
+            /// }
+            /// void Callee(int a, int b, params bool[] c)
+            /// {
+            ///     DoSomething(a, b, c);
+            /// }
+            /// 4. A special case, the parameter is only read once in the callee method body
+            /// Before:
+            /// void Caller(bool x)
+            /// {
+            ///     Callee(Foo(), Bar())
+            /// }
+            /// void Callee(int a, int b)
+            /// {
+            ///     DoSomething(a, b);
+            /// }
+            /// After:
+            /// void Caller(bool x)
+            /// {
+            ///     DoSomething(Foo(), Bar());
+            /// }
+            /// void Callee(int a, int b)
+            /// {
+            ///     DoSomething(a, b);
+            /// }
+            /// In this case, parameters 'a' and 'b' should just be replaced by the argument expression.
+            /// Note: this might cause semantics changes. It is by design.
+            /// </summary>
+            public ImmutableDictionary<IParameterSymbol, TExpressionSyntax> ParametersToReplace { get; }
+
+            public MethodParametersInfo(
+                ImmutableArray<(IParameterSymbol parameterSymbol, string name)> parametersWithVariableDeclarationArgument,
+                ImmutableArray<(IParameterSymbol parameterSymbol, TExpressionSyntax expression)> parametersToGenerateFreshVariablesFor,
+                ImmutableDictionary<IParameterSymbol, TExpressionSyntax> parametersToReplace)
+            {
+                ParametersWithVariableDeclarationArgument = parametersWithVariableDeclarationArgument;
+                ParametersToGenerateFreshVariablesFor = parametersToGenerateFreshVariablesFor;
+                ParametersToReplace = parametersToReplace;
+            }
+        }
+
+        private async Task<MethodParametersInfo> GetMethodParametersInfoAsync(
+            Document document,
```

Comment:
1. since you have a caller and callee, `document` is pretty ambiguous.  Maybe `invocationDocument`.
2. i just realized, how do you deal with cross-language inline-method?  is it expliclty blocked?  do you crash?  do you have a test?
3. do you ahve tests with same-language, but different documents for the methods?

Reply:
I think this feature is limited to private method. So when I write this I think I have taken the partial class into account.
Is that possible to create partial class from different language? (I tested in VS look like it can't)
But I think you are right, it needs some cross document tests

Reply:
also, does anything validate that the acller of the private method is in the same type that the private method is declared in?

also, you can access privates from other nested types of a type :)   all good cases to at least test.

Reply:
Test with cross language scenario, 
I have a C# project that referencing a VB project, then in C# class create a VB class instance. In try to inline the private method.
![image](https://user-images.githubusercontent.com/24360909/91622779-a0ef1c80-e94d-11ea-893e-5f4ee2c4748f.png)
 And it can't get the VB method symbol
![image](https://user-images.githubusercontent.com/24360909/91622838-e0b60400-e94d-11ea-8695-621dfed84e9f.png)
So this scenario will be filtered. :)

Reply:
great.  def add test though :)

Reply:
Test added. Marked as Resolved.